annotated_test.go

```diff
@@ -1474,6 +1474,159 @@ func TestAnnotate(t *testing.T) {
+			giveAnnotation: fx.ParamTags(`name:"something'`),
+		},
+		{
+			give:           "ParamTags wrong starting quote",
+			wantErr:        errTagValueSyntaxQuote + "`optional:\"true\"` ?",
+			giveAnnotation: fx.ParamTags(`name:"something" optional:'true"`),
+		},
+		{
+			give:           "ParamTags key not made of alphabets",
+			wantErr:        errTagKeySyntax,
+			giveAnnotation: fx.ParamTags(`a"me1:"something"`),
+		},
+		{
+			give:           "ParamTags multiple tags not separated by space",
+			wantErr:        errTagSyntaxSpace,
+			giveAnnotation: fx.ParamTags(`name:"something"group:"something"`),
+		},
+		{
+			give:           "ParamTags key value not separated by colon",
+			wantErr:        errTagSyntaxColon + "`group:\"something\"` ?",
+			giveAnnotation: fx.ParamTags(`group"something"`),
+		},
+		{
+			give:           "ParamTags key not equal to group, name or optional",
+			wantErr:        errTagKeySyntax,
+			giveAnnotation: fx.ParamTags(`name1:"something"`),
+		},
+		{
+			give:           `ParamTags invalid space after key colon`,
+			wantErr:        errTagValueSpace,
+			giveAnnotation: fx.ParamTags(`name: "something"`),
+		},
+		{
+			give:           "ResultTags wrong ending quote",
+			wantErr:        errTagValueSyntaxEndingQuote + "`name:\"something\"` ?",
+			giveAnnotation: fx.ResultTags(`name:"something'`),
+		},
+		{
+			give:           "ResultTags wrong starting quote",
+			wantErr:        errTagValueSyntaxQuote + "`optional:\"true\"` ?",
+			giveAnnotation: fx.ResultTags(`name:"something" optional:'true"`),
+		},
+		{
+			give:           "ResultTags key not made of alphabets",
+			wantErr:        errTagKeySyntax,
+			giveAnnotation: fx.ResultTags(`a"me1:"something"`),
+		},
+		{
+			give:           "ResultTags multiple tags not separated by space",
+			wantErr:        errTagSyntaxSpace,
+			giveAnnotation: fx.ResultTags(`name:"something"group:"something"`),
+		},
+		{
+			give:           "ResultTags key value not separated by colon",
+			wantErr:        errTagSyntaxColon + "`group:\"something\"` ?",
+			giveAnnotation: fx.ResultTags(`group"something"`),
+		},
+		{
+			give:           "ResultTags key not equal to group, name or optional",
+			wantErr:        errTagKeySyntax,
+			giveAnnotation: fx.ResultTags(`name1:"something"`),
+		},
+		{
+			give:           "ResultTags no key",
+			wantErr:        errTagKeySyntax,
+			giveAnnotation: fx.ResultTags(`:"something"`),
+		},
+		{
+			give:           "ResultTags invalid space after key colon ",
+			wantErr:        errTagValueSpace,
+			giveAnnotation: fx.ResultTags(`name: "something"`),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.give, func(t *testing.T) {
+			app := NewForTest(t,
+				fx.Provide(
+					fx.Annotate(
+						newA,
+						tt.giveAnnotation,
+					),
+				),
+				fx.Invoke(newB),
+			)
+			err := app.Err()
+			assert.ErrorContains(t, err, tt.wantErr)
+		})
+	}
+
+	tests1 := []struct {
```

Comment:
Let's put two separate sub-tests, one for your valid cases and one for your invalid cases